build-tools/packages/build-cli/src/lib/release.ts

```diff
@@ -0,0 +1,127 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+import {
+    ReleaseVersion,
+    VersionBumpType,
+    VersionScheme,
+    detectVersionScheme,
+    getVersionRange,
+} from "@fluid-tools/version-tools";
+
+import { ReleaseGroup } from "../releaseGroups";
+
+export interface PackageVersionList {
+    [packageName: string]: string;
+}
+
+export interface ReleaseReport {
+    [packageName: string]: ReleaseDetails;
+}
+
+export interface ReleaseDetails {
+    version: ReleaseVersion;
+    previousVersion?: ReleaseVersion;
+    versionScheme: VersionScheme;
+    date?: Date;
+    releaseType: VersionBumpType;
+    isNewRelease: boolean;
+    releaseGroup?: ReleaseGroup;
+    ranges: ReleaseRanges;
+}
+
+export interface ReleaseRanges {
```

Comment:
A doc comment on this noting why we need title and carrot separate from minor and patch would help readers understand this code (I understand why our internal packages break title and carrot, but I don't get why we care about them enough to include them here if you could just use minor and patch which actually work instead)

Reply:
Fixed in latest iteration.